app/policies/user_policy.rb

```diff
@@ -0,0 +1,2 @@
+class UserPolicy < ApplicationPolicy
+end
```

Comment:
We talked about RBAC and I think you want to implement it with Pundit by specifying what given role can do every resource. Let's take a look at this example table - tables like that are often used to represent RBAC rules:
![image](https://user-images.githubusercontent.com/36715675/191609347-e07a4af4-43a4-420e-9f54-75498fc2b0d3.png)

Looking at this, you are trying to implement RBAC by translating columns into policies. It is an working approach, but not one that Pundit was built for and I would argue it is much harder approach to maintain that it needs to be. I would suggest you to implement policies as rows, so instead of `UserPolicy`, `ManagerPolicy` and `AdminPolicy` you would have `HallPolicy`, `MoviePolicy` - etc. That way you will end up with smaller policies, that are simpler to maintain - and you will not have to fight with Pundit, as this is default way of using it 😉